### PR TITLE
Enhance `Opposite`-tests

### PR DESCRIPTION
**Pull request by TKuh**
Note: this should make `LazyCategories` fail due to #1295 and succeed after PR https://github.com/homalg-project/CAP_project/pull/1297.

### CODE REVIEW

File: CAP/gap/OppositeCategory.gi [hunk 1]
```diff
@@ -38,6 +38,27 @@ InstallMethod( Opposite,
...
 end );
 
+##
+InstallOtherMethod( Opposite,
+                    [ IsCapCategory, IsCapCategoryMorphism ],
+                    
+  function( opposite_cat, morphism )
+    local source, range;
+    
+    if not HasOpposite( opposite_cat ) then
```
Comment (zickgraf) [inline CAP/gap/OppositeCategory.gi:48]: I think this check is not what we want: If you compute `Opposite( C )`, the also `C` as the attribute `Opposite`. You could check for `WasCreatedAsOppositeCategory`, but I would suggest to simply remove this check and rely on `ObjectConstructor` and `MorphismConstructor` for checks.

**Review by zickgraf — CHANGES_REQUESTED** [review]

**Review by zickgraf — APPROVED** [review]
Thanks a lot!

**Comment by zickgraf** [comment]
Test run in CategoricalTowers: https://github.com/homalg-project/CategoricalTowers/actions/runs/4917003133 Let's hope it fails :D

**Comment by zickgraf** [comment]
FinSetsForCAP is already failing: https://github.com/homalg-project/FinSetsForCAP/actions/runs/4917118682/jobs/8781786322 :-)

@TKuh Please rebase #1297.

**Comment by mohamed-barakat** [comment]
Wonderful, thank you very much @TKuh.

**Comment by zickgraf** [comment]
LazyCategories is also failing :-)

**Pull request merged**